app/models/concerns/host_params.rb

```diff
@@ -8,59 +8,65 @@ module HostParams
     include ParameterValidators
     attr_reader :cached_host_params
 
-    def params
-      host_params.update(lookup_keys_params)
-    end
+    register_info_provider :host_params_info
+  end
 
-    def clear_host_parameters_cache!
-      @cached_host_params = nil
-    end
+  def host_params_info
+    { 'parameters' => params }
+  end
 
-    def host_inherited_params(include_source = false)
-      hp = {}
-      params = host_inherited_params_objects
-      params.each do |param|
-        source = param.associated_type
-        options = {:value => param.value,
-                   :source => source,
-                   :safe_value => param.safe_value }
-        if source != 'global'
-          options.merge!(:source_name => param.associated_label)
-        end
-        hp.update(Hash[param.name => include_source ? options : param.value])
-      end
-      hp
-    end
+  def params
+    host_params
+  end
 
-    def host_params
-      return cached_host_params unless cached_host_params.blank?
-      hp = host_inherited_params
-      # and now read host parameters, override if required
-      host_parameters.authorized(:view_params).each { |p| hp[p.name] = p.value }
-      @cached_host_params = hp
-    end
+  def clear_host_parameters_cache!
+    @cached_host_params = nil
+  end
 
-    def host_inherited_params_objects
-      params = CommonParameter.all
-      if SETTINGS[:organizations_enabled] && organization
-        params += extract_params_from_object_ancestors(organization)
+  def host_inherited_params(include_source = false)
+    hp = {}
+    params = host_inherited_params_objects
+    params.each do |param|
+      source = param.associated_type
+      options = {:value => param.value,
+                 :source => source,
+                 :safe_value => param.safe_value }
+      if source != 'global'
+        options.merge!(:source_name => param.associated_label)
       end
+      hp.update(Hash[param.name => include_source ? options : param.value])
+    end
+    hp
+  end
 
-      if SETTINGS[:locations_enabled] && location
-        params += extract_params_from_object_ancestors(location)
-      end
+  def host_params
+    return cached_host_params unless cached_host_params.blank?
+    hp = host_inherited_params
+    # and now read host parameters, override if required
+    host_parameters.authorized(:view_params).each { |p| hp[p.name] = p.value }
+    @cached_host_params = hp
```

Comment:
You are right with the analysis. My consideration is that in step 2 the host object is still in-memory only (if the user cancels the change or just closes the window - the change is not applied), which means that the cache shouldn't be updated.
I can see only two cases when the cache would be effective:
1. For multiple access to parameters on the same instance
2. Read-only access to the host from multiple requests.
Anyway this cache is there for too long (I have found the [commit](https://github.com/theforeman/foreman/commit/8ea536e3cf5cb9566953dc1772973160a99eb41f) where it is added). Maybe it's time to get rid of it... If you want, I'll open a new issue in Redmine to remind us to do that.